### DISCUSSION TITLE
Garmin Two-Step Verification Token?

### DISCUSSION THREAD

**Discussion by geoffbon** [philosowaffle/peloton-to-garmin#570] [General]
Is it possible to store the MFA token from the initial login?  Would be great to enter the code once, then leave it set to sync regularly in the background.  I'm using the local Windows exe, but would happily move to docker / web UI / whatever.
Looks like the Withings sync project came over here to find out how to do Garmin MFA, but also solved the token storage issue, at least if I'm understanding this comment:
https://github.com/jaroslawhartman/withings-sync/issues/43#issuecomment-1752211438

Edit: I see they use the [Garth library](https://github.com/matin/garth) to solve this, feasible to do the same?

**Comment by geoffbon** [comment]
Is it worth me creating an issue for this?

**Reply by philosowaffle** [reply]
This is something that's been floating at the back of mind ever since I rewrote auth a few months back.  P2G can't directly use the Garth library, but I followed their pattern and translated what was needed into P2G (#524).

Just created an issue here to track this officially: https://github.com/philosowaffle/peloton-to-garmin/issues/585
